action/action.yml

```diff
@@ -33,7 +33,7 @@ runs:
         fi
 
         if [ -z "${VERSION}" ]; then
-          VERSION=$(curl https://api.github.com/repos/fluxcd/flux2/releases/latest -sL "${TOKEN[@]}" | grep tag_name | sed -E 's/.*"([^"]+)".*/\1/' | cut -c 2-)
+          VERSION=$(curl https://api.github.com/repos/fluxcd/flux2/releases/latest -sL ${TOKEN[@]} | grep tag_name | sed -E 's/.*"([^"]+)".*/\1/' | cut -c 2-)
```

Comment:
Why do we write `TOKEN[@]` instead of just `TOKEN` here? The "TOKEN" variable is not an array.

Reply:
`TOKEN=(-H "Authorization: token ${TOKEN}")` is this not an array definition in bash?

Reply:
The failure seems to be related to `set -o pipefail` which is another option I didn't realize is in effect.

(Any idea what "exit code 7" from `cut` means?)

Reply:
Sorry, the output was misleading, that is not an exit code from `cut -d 2-` which removes the `v` from the front of the version number. It's error code 7 from `curl`.

```
curl -v https://api.github.com/repos/fluxcd/flux2/releases/latest -sL -H "Authorization: token gho_abcdSECRET"
```

This is the structure of the command we need to run, which is why the construction `TOKEN=(-H "Authorization: token ${TOKEN}")` was used in the first place I guess, because it lets us pass those around together.

```
curl -v https://api.github.com/repos/fluxcd/flux2/releases/latest -sL -H Authorization: token gho_abcdSECRET
```
This will fail because the authorization token header string is not all considered an argument to `-H`.

I think the safest thing to do now is to just have two branches, since it's clear enough to me what's failing and clear ho to fix, and I won't be happy with the branch conditional, but I'm open to suggestions if anyone wants to suggest a cleaner refactor.

I will open up another PR and we can continue the discussion there.

Reply:
* https://github.com/fluxcd/flux2/pull/3513

I believe this fixes it. Either pass a quoted parameter pair, if there is one – or pass no parameter. (Do not pass an empty string, or unquoted parameter.)

If there is a better structure for doing that pattern in bash, I'm all ears.